src/serial_bridge_index.cpp

```diff
@@ -134,7 +135,12 @@ std::string serial_bridge::decompress(const char *buffer, size_t length) {
         throw std::runtime_error("Incomplete decompression: more data was expected");
     }
 
-	return decompressedData;
+    // Allocate memory for the return value
+    char* result = new char[decompressedData.size() + 1];
```

Comment:
why not `std::vector`?

Reply:
I considered the performance criteria when choosing char*, it might be marginally faster because there is no abstraction overhead, but the difference seems to be minimal. I will make another version using std::vector.

Reply:
Updated at PR: https://github.com/ExodusMovement/mymonero-core-cpp/pull/36